core/modules/qproc/QuerySession.h

```diff
@@ -79,7 +78,17 @@ class QuerySession {
 
     std::string const& getOriginal() const { return _original; }
     void setDefaultDb(std::string const& db);
-    void setQuery(std::string const& q);
+
+    /**
+     * @brief Analyze SQL query issued by user
+     *
+     * This query comes from user through mysql-client and mysql-proxy
+     * This function will parse it, apply query plugins (i.e. build parallel and
+     * merge queries) and check for errors
+     *
+     * @param sql: the sql query
+     */
+    void analyzeQuery(std::string const& sql);
```

Comment:
This method is much more than a simple setter: there's no "query" member in QuerySession, and the method parse the sql input, apply query plugins to generate parallel and merge query and may returns lots of kind of error. So I would like a more explicit name than setQuery (one could think it's a trivial setter). Maybe setQueryExecutionPlan()?